src/epac/flatbuffers/pva0_data.py

```diff
@@ -0,0 +1,979 @@
+# - `_serialise_scalar`: Serializes a single scalar, supporting various types.
+# - `_serialise_string_array`: Serializes an array of strings.
+# - `_serialise_array`: Serializes an array of values, supporting various types.
+#
+# **Error Handling**:
+# - If the data type is unsupported, a `TypeError` is raised for scalar and array types.
+
+
+def _serialise_string_scalar(builder: flatbuffers.Builder, data: np.ndarray):
+    return builder.CreateString(data.item()), String, AnyInner.AnyInner.String
+
+
+def _serialise_scalar(builder: flatbuffers.Builder, data: np.ndarray):
+    try:
+        return (
+            data.item(),
+            map_dtype_to_scalar_fb[data.dtype],
+            map_dtype_to_any_scalar_enum[data.dtype],
+        )
+    except KeyError:
+        raise TypeError(f"Unsupported scalar dtype: {data.dtype}")
+
+
+def _serialise_string_array(builder: flatbuffers.Builder, data: np.ndarray):
+    internal_values_offsets = [builder.CreateString(item) for item in reversed(data)]
+    StringArray.StringArrayStartValueVector(builder, len(data))
+    for start_offset in internal_values_offsets:
+        builder.PrependSOffsetTRelative(start_offset)
+    return builder.EndVector(), StringArray, AnyInner.AnyInner.StringArray
+
+
+def _serialise_array(builder: flatbuffers.Builder, data: np.ndarray):
+    try:
+        return (
+            builder.CreateNumpyVector(data),
+            map_dtype_to_array_fb[data.dtype],
+            map_dtype_to_any_array_enum[data.dtype],
+        )
+    except KeyError:
+        raise TypeError(f"Unsupported array dtype: {data.dtype}")
+
+
+@safe_serialise
+def serialise_any(builder: flatbuffers.Builder, data) -> int:
+    """serialises an arbitrary object into a FlatBuffers Any union.
+
+    Args:
+        builder (flatbuffers.Builder): The FlatBuffers builder used to construct the object.
+        data: The object to be serialised, which must be compatible with the Any union.
+
+    Returns:
+        The FlatBuffers offset for the serialised Any object.
+
+    Raises:
+        TypeError: If an attempt is made to serialize a multi-dimensional array (ndim > 1).
+    """
+    data = np.asarray(data)
+    dtype = data.dtype
+
+    if data.ndim == 0:  # Scalar
+        if np.issubdtype(data.dtype, np.str_) or np.issubdtype(data.dtype, np.bytes_):
+            data_start_offset, fb_type, enum_type = _serialise_string_scalar(
+                builder, data
+            )
+        else:
+            data_start_offset, fb_type, enum_type = _serialise_scalar(builder, data)
+    elif data.ndim == 1:  # Array
+        if np.issubdtype(dtype, np.str_) or np.issubdtype(dtype, np.bytes_):
+            data_start_offset, fb_type, enum_type = _serialise_string_array(
+                builder, data
+            )
+        else:
+            data_start_offset, fb_type, enum_type = _serialise_array(builder, data)
+    else:
+        raise TypeError("Ony 0D and 1D Arrays are supported.")
```

Comment:
Nitpicks:

```suggestion
        raise TypeError("only scalars and 1D Arrays are supported")
```

I think the usual convention for exception messages is that they start with a lowercase letter and don't end with a `.`

Reply:
ok will correct this and an others I find